Calibration.py

```diff
@@ -200,41 +176,24 @@ def write_config(self):
         except:
             print("Failed to add, could already exist")
 
-        # Write out the wheel motor ESC ranges
-        config.set('motors', 'LEFT_MIN', str(self.core.left_servo.servo_min))
-        config.set('motors', 'LEFT_MID', str(self.core.left_servo.servo_mid))
-        config.set('motors', 'LEFT_MAX', str(self.core.left_servo.servo_max))
-
-        config.set('motors', 'RIGHT_MIN', str(self.core.right_servo.servo_min))
-        config.set('motors', 'RIGHT_MID', str(self.core.right_servo.servo_mid))
-        config.set('motors', 'RIGHT_MAX', str(self.core.right_servo.servo_max))
-
-        # Write out the Auxilery ESC ranges
-        config.set(
-            'motors',
-            'LEFT_AUX_1_MIN',
-            str(self.core.left_aux_1_servo.servo_min))
-        config.set(
-            'motors',
-            'LEFT_AUX_1_MID',
-            str(self.core.left_aux_1_servo.servo_mid))
-        config.set(
-            'motors',
-            'LEFT_AUX_1_MAX',
-            str(self.core.left_aux_1_servo.servo_max))
-        # And now the right
-        config.set(
-            'motors',
-            'RIGHT_AUX_1_MIN',
-            str(self.core.right_aux_1_servo.servo_min))
-        config.set(
-            'motors',
-            'RIGHT_AUX_1_MID',
-            str(self.core.right_aux_1_servo.servo_mid))
-        config.set(
-            'motors',
-            'RIGHT_AUX_1_MAX',
-            str(self.core.right_aux_1_servo.servo_max))
+        # Write out min/mid/max of menu items that we can change.
+        for item in self.menu_list.items():
+            item_enum_str = str(item[0]).split('.')
+            item_min_str = "{}_MIN".format(item_enum_str[0])
+            item_mid_str = "{}_MID".format(item_enum_str[0])
+            item_max_str = "{}_MAX".format(item_enum_str[0])
+            config.set(
+                'motors',
+                item_min_str,
+                str(self.core.servos[item[0]][0].servo_min))
+            config.set(
+                'motors',
+                item_mid_str,
+                str(self.core.servos[item[0]][0].servo_mid))
+            config.set(
+                'motors',
+                item_max_str,
+                str(self.core.servos[item[0]][0].servo_max))
```

Comment:
I think you can ditch some of this boilerplate with judicious use of `zip`, and a bit of `getattr`:
```
for item in self.menu_list.items():
    item_enum_str = str(item[0]).split('.')[0]
    for (suffix, prop) in zip(('MIN', 'MID', 'MAX'), ('servo_min', 'servo_mid', 'servo_max')):
        config_set(
            'motors',
            '{0}_{1}'.format(item_enum_str, suffix),
            str(getattr(self.core.servos[item[0]][0], prop))
        )
```